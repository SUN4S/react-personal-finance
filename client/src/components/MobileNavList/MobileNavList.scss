.mobileNav {
  position: fixed;
  bottom: 0;
  height: 50px;
  width: 100%;
  border-top: 2px solid var(--page-bg);
  background-color: var(--sidebar-bg);
  z-index: 10;

  ul {
    margin-top: 0;
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: space-around;

    li {
      width: 48px;
      height: 48px;
      display: flex;
      align-items: center;
      justify-content: center;
      border-radius: 3px;
      background-color: none;

      &.buttonContainer {
        width: 25%;
        height: 48px;
        border: 2px solid var(--primary);
        border-radius: 3px;

        button {
          height: 100%;
          width: 100%;
          display: flex;
          align-items: center;
          justify-content: center;
          background-color: transparent;
          border: none;
          color: var(--primary);
          font-size: 40px;

          &:hover {
            cursor: pointer;
            background-color: var(--page-bg);
          }

          &:active {
            border: 3px solid var(--primary);
          }
        }
      }

      a {
        height: 48px;
        width: 48px;
        display: flex;
        align-items: center;
        justify-content: center;
        padding: 0;

        .logoContainer {
          width: 40px;
          height: 40px;
          display: flex;
          align-items: center;
          justify-content: center;

          svg {
            height: 25px;
            width: 25px;

            .cls-1,
            .cls-2,
            .cls-3,
            text {
              stroke: var(--text-secondary);
              transition: stroke 0.2s ease-in-out;
            }
          }
        }

        &.active {
          background-color: transparent;

          svg {
            stroke: var(--primary);
            animation: linkSelect 0.3s ease-out;

            .cls-1,
            .cls-2,
            .cls-3,
            text {
              stroke: var(--primary);
            }
          }
        }
      }
    }
  }
}

@keyframes linkSelect {
  0% {
    transform: scale(1.2);
  }
  100% {
    transform: scale(1);
  }
}
