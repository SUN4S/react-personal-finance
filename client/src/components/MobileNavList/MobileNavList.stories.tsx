import { Meta, Story } from "@storybook/react/types-6-0";

import { MobileNavList } from "./MobileNavList";

interface TemplateProps {
  theme: string;
  onClick: Function;
}

export default {
  title: "Components/Mobile Nav List",
  component: MobileNavList,
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  return (
    <html data-theme={args.theme}>
      <body
        style={{
          backgroundColor: "var(--sidebar-bg)",
        }}
      >
        <MobileNavList clickFunction={args.onClick} />
      </body>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  onClick: () => alert("Clicked"),
};

export const Default_Dark = Template.bind({});
Default_Dark.args = {
  theme: "dark",
  onClick: () => alert("Clicked"),
};
