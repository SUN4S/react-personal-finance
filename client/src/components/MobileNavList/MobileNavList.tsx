import "./MobileNavList.scss";

import { IconDashboard } from "../Icons/IconDashboard/IconDashboard";
import { IconHistory } from "../Icons/IconHistory/IconHistory";
import { IconStock } from "../Icons/IconStock/IconStock";
import { IconWeekly } from "../Icons/IconWeekly/IconWeekly";
import { NavLink } from "react-router-dom";

export const MobileNavList = (props: { clickFunction: Function }) => {
  return (
    <div className="mobileNav">
      <ul>
        <li>
          <NavLink to="/">
            <div className="logoContainer">
              <IconDashboard />
            </div>
          </NavLink>
        </li>
        <li>
          <NavLink to="/history">
            <div className="logoContainer">
              <IconHistory />
            </div>
          </NavLink>
        </li>
        <li className="buttonContainer">
          <button onClick={props.clickFunction()}>+</button>
        </li>
        <li>
          <NavLink to="/stocks">
            <div className="logoContainer">
              <IconStock />
            </div>
          </NavLink>
        </li>
        <li>
          <NavLink to="/weekly">
            <div className="logoContainer">
              <IconWeekly />
            </div>
          </NavLink>
        </li>
      </ul>
    </div>
  );
};
