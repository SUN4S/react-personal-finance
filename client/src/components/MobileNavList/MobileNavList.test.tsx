import { cleanup, fireEvent, render } from "@testing-library/react";

import { MemoryRouter } from "react-router-dom";
import { MobileNavList } from "./MobileNavList";
import renderer from "react-test-renderer";

describe("Mobile Navigation List", () => {
  afterEach(cleanup);
  const mockFunction = jest.fn();

  it("Renders", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={[{ pathname: "/" }]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    expect(getByTestId("mobileNavList")).toBeTruthy();
  });

  it("Renders Link List", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={[{ pathname: "/" }]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    expect(getByTestId("mobileNavList").querySelectorAll("ul li").length).toBe(
      5
    );
  });

  it("Has Aria-label", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={[{ pathname: "/" }]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );

    expect(getByTestId("mobileNavListButton")).toHaveAttribute(
      "aria-label",
      "Mobile Navigation Button"
    );
  });

  it("Calls Function", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={[{ pathname: "/" }]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    fireEvent.click(getByTestId("mobileNavListButton"));
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });

  it("Dashboard Link Exists", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/currentUri"]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    // navigate here on event
    expect(
      getByTestId("mobileNavList").querySelectorAll("ul li a")[0]
    ).toHaveAttribute("href", "/");
  });

  it("History Link Exists", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/currentUri"]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    // navigate here on event
    expect(
      getByTestId("mobileNavList").querySelectorAll("ul li a")[1]
    ).toHaveAttribute("href", "/history");
  });

  it("Stocks Link Exists", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/currentUri"]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    // navigate here on event
    expect(
      getByTestId("mobileNavList").querySelectorAll("ul li a")[2]
    ).toHaveAttribute("href", "/stocks");
  });

  it("Weekly Link Exists", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/currentUri"]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    // navigate here on event
    expect(
      getByTestId("mobileNavList").querySelectorAll("ul li a")[3]
    ).toHaveAttribute("href", "/weekly");
  });

  it("Matches Snapshot", () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={["/currentUri"]}>
        <MobileNavList clickFunction={mockFunction} />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
