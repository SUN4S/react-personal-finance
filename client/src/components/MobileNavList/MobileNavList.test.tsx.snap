// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Navigation List Matches Snapshot 1`] = `
<div
  className="mobileNav"
  data-testid="mobileNavList"
>
  <ul>
    <li>
      <a
        className=""
        href="/"
        onClick={[Function]}
      >
        <div
          className="logoContainer"
        >
          <svg
            id="dashboardIcon"
            viewBox="0 0 52 52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Layer 1"
                id="Layer_1-2"
              >
                <rect
                  className="cls-1"
                  height="50"
                  rx="6"
                  width="50"
                  x="1"
                  y="1"
                />
                <rect
                  className="cls-2"
                  height="13.91"
                  rx="3"
                  width="17.13"
                  x="28.59"
                  y="30.39"
                />
                <rect
                  className="cls-2"
                  height="13.91"
                  rx="3"
                  width="17.13"
                  x="6.13"
                  y="30.39"
                />
                <rect
                  className="cls-2"
                  height="13.91"
                  rx="3"
                  width="17.13"
                  x="28.67"
                  y="8.85"
                />
                <rect
                  className="cls-2"
                  height="13.91"
                  rx="3"
                  width="17.13"
                  x="6.2"
                  y="8.85"
                />
              </g>
            </g>
          </svg>
        </div>
      </a>
    </li>
    <li>
      <a
        className=""
        href="/history"
        onClick={[Function]}
      >
        <div
          className="logoContainer"
        >
          <svg
            id="historyIcon"
            viewBox="0 0 52 52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Layer 1"
                id="Layer_1-2"
              >
                <rect
                  className="cls-1"
                  height="50"
                  rx="6"
                  width="50"
                  x="1"
                  y="1"
                />
                <line
                  className="cls-2"
                  x1="27.33"
                  x2="27.33"
                  y1="28.95"
                  y2="17.36"
                />
                <line
                  className="cls-2"
                  x1="27.33"
                  x2="33.25"
                  y1="28.95"
                  y2="31.59"
                />
                <path
                  className="cls-2"
                  d="M16.84,39.26a17.53,17.53,0,1,0-6-12"
                />
                <line
                  className="cls-2"
                  x1="11.41"
                  x2="16.84"
                  y1="27.16"
                  y2="21.72"
                />
                <line
                  className="cls-2"
                  x1="6.21"
                  x2="10.05"
                  y1="20.06"
                  y2="26.72"
                />
              </g>
            </g>
          </svg>
        </div>
      </a>
    </li>
    <li
      className="buttonContainer"
    >
      <button
        aria-label="Mobile Navigation Button"
        data-testid="mobileNavListButton"
        onClick={[Function]}
      >
        +
      </button>
    </li>
    <li>
      <a
        className=""
        href="/stocks"
        onClick={[Function]}
      >
        <div
          className="logoContainer"
        >
          <svg
            id="historyIcon"
            viewBox="0 0 52 52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Layer 1"
                id="Layer_1-2"
              >
                <rect
                  className="cls-1"
                  height="50"
                  rx="6"
                  width="50"
                  x="1"
                  y="1"
                />
                <line
                  className="cls-2"
                  x1="14.16"
                  x2="24.03"
                  y1="36.51"
                  y2="24.56"
                />
                <line
                  className="cls-2"
                  x1="24.03"
                  x2="30.25"
                  y1="24.56"
                  y2="32.68"
                />
                <line
                  className="cls-2"
                  x1="30.25"
                  x2="42.13"
                  y1="32.68"
                  y2="13.23"
                />
                <line
                  className="cls-2"
                  x1="42.74"
                  x2="37.59"
                  y1="11.79"
                  y2="13.18"
                />
                <line
                  className="cls-2"
                  x1="44.13"
                  x2="42.74"
                  y1="16.93"
                  y2="11.79"
                />
                <line
                  className="cls-3"
                  x1="8.69"
                  x2="8.69"
                  y1="8.49"
                  y2="43.51"
                />
                <line
                  className="cls-3"
                  x1="43.71"
                  x2="8.69"
                  y1="43.51"
                  y2="43.51"
                />
              </g>
            </g>
          </svg>
        </div>
      </a>
    </li>
    <li>
      <a
        className=""
        href="/weekly"
        onClick={[Function]}
      >
        <div
          className="logoContainer"
        >
          <svg
            id="weeklyIcon"
            viewBox="0 0 52 52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Layer 1"
                id="Layer_1-2"
              >
                <rect
                  className="cls-1"
                  height="50"
                  rx="6"
                  width="50"
                  x="1"
                  y="1"
                />
                <text
                  className="cls-2"
                  transform="translate(15 41)"
                >
                  7
                </text>
              </g>
            </g>
          </svg>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
